src/main/java/no/difi/statistics/elasticsearch/commands/CategoriesQuery.java

```diff
@@ -0,0 +1,108 @@
+package no.difi.statistics.elasticsearch.commands;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import no.difi.statistics.model.OwnerCategories;
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.RestClient;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CategoriesQuery {
+
+    private RestClient elasticSearchClient;
+
+    private CategoriesQuery() {
+    }
+
+    public Set<OwnerCategories> execute() throws IOException {
+        // Put index-name and categories in Map, and copy to a Set before returning.
+        Map<String, OwnerCategories> ownerCategoriesMap = new HashMap<>();
+
+        Request request = new Request("GET", "/*@*@*/_mapping");
+        String mappings = EntityUtils.toString(elasticSearchClient.performRequest(request).getEntity());
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode jsonNode = mapper.readTree(mappings);
+        Iterator<String> index = jsonNode.fieldNames();
+        for (JsonNode mappingsOfIndex : jsonNode) {
+            // Example of an index-name: 991825827@idporten-innlogging@hour2022
+            String[] indexNameTokens = index.next().split("@", 3);
+            String owner = indexNameTokens[0];
+            String name = indexNameTokens[1];
+            String distance = determineDistance(indexNameTokens[2]);
+            String key = owner + ":" +  name + ":" + distance;
```

Comment:
"Før i tiden" strakk man seg langt for å unngå streng-konkatenering med + i en loop (siden Strings er immutable, må man lage en ny streng hver gang..), men nå kan det hende kompiler bytter det ut for deg, eller du kan f. eks. bruke eksplisitt en StringBuilder.

Reply:
Yes det var sånn før i tiden, men nå er det ikke slik om jeg har forstått ting riktig. Googlet litt for å finne backing på det og om du leser svaret i denne Stack-overflowen`en så er begrunnelsen godt forklart: https://stackoverflow.com/a/1532499